avocado/utils/vmimage.py

```diff
@@ -218,8 +218,11 @@ def get_image_url(self):
         if self.url_old_images and int(self.version) <= 38:
             self.url_versions = self.url_old_images
 
-        self.url_images = self.url_versions + "{version}/" + cloud + "/{arch}/images/"
-        return super().get_image_url()
+        self.url_images = (
```

Comment:
This block seems unrelated (unnecessary) to the actual fix.  In fact, it changes a bit the expectations of the `url_versions` attribute, because it's supposed to contain a format string until the very last minute when `get_image_url()` is called.

IMO, it's better to remove this (assuming, and based on my testing, that this is not necessary for the fix to be effective).